src/hoc/with-analytic-on-event.tsx

```diff
@@ -1,126 +1,82 @@
+    EventType extends AnyFn,
+    BaseProps extends Record<EventName, EventType> = Record<EventName, EventType>
+>({
     eventName,
     analyticName,
-    extras,
-    identities,
-}: WithAnalyticOnEventConfiguration<Props, Event>) => (BaseComponent: ElementType<Props>) => {
-    type CombinedProps = Props & WithAnalyticOnEventProps<Event>;
+    extendAnalytics: extendAnalyticsFromConfig,
+}: WithAnalyticOnEventConfiguration<EventName, EventType, BaseProps>) => <Props extends BaseProps = BaseProps>(
+    BaseComponent: ElementType<Props>,
+) => {
+    type CombinedProps = Omit<Props, EventName> &
+        Partial<Record<EventName, Props[EventName]>> &
+        WithAnalyticOnEventProps<Parameters<Props[EventName]>>;
 
     const EnhancedComponent: FunctionComponent<CombinedProps> = ({
         [eventName]: rawEvent,
-        analyticsExtras,
-        analyticsIdentities,
+        extendAnalytics,
         shouldDispatchAnalytics,
         ...props
-    }) => (
-        <ShisellContext.Consumer>
-            {(analytics) => (
-                <AnalyticOnEvent
-                    analytics={analytics}
-                    event={(rawEvent as unknown) as Function}
-                    eventName={eventName as string}
-                    analyticName={analyticName}
-                    analyticsExtras={analyticsExtras}
-                    analyticsIdentities={analyticsIdentities}
-                    shouldDispatchAnalytics={shouldDispatchAnalytics}
-                    staticExtras={extras}
-                    staticIdentities={identities}
-                    displayName={EnhancedComponent.displayName!}
-                >
-                    {(event) => {
-                        // @ts-ignore
-                        return <BaseComponent {...{[eventName]: event}} {...props} />;
-                    }}
-                </AnalyticOnEvent>
-            )}
-        </ShisellContext.Consumer>
-    );
+    }) => {
+        const analytics = useAnalytics();
+
+        const onEvent = useCallback(
+            (...args: Parameters<Props[EventName]>) => {
+                const shouldDispatch =
+                    shouldDispatchAnalytics !== false &&
+                    (typeof shouldDispatchAnalytics !== 'function' || shouldDispatchAnalytics(...args));
+
+                if (shouldDispatch) {
+                    let {dispatcher} = analytics;
+
+                    if (extendAnalyticsFromConfig) {
+                        dispatcher = dispatcher.extend(extendAnalyticsFromConfig(props as any, ...args));
+                    }
+                    if (extendAnalytics) {
+                        dispatcher = dispatcher.extend(extendAnalytics(...args));
+                    }
+                    dispatcher.dispatch(analyticName);
+                }
+
+                if (typeof rawEvent === 'function') {
+                    return rawEvent(...args);
+                } else if (process.env.NODE_ENV !== 'prodution' && rawEvent) {
```

Comment:
```suggestion
                } else if (process.env.NODE_ENV !== 'production' && rawEvent) {
```

Reply:
The `yarn build` output contains the `process.env.NODE_ENV` check.
Is that expected? Normally, I'd see this tree shaken out or only kept in a `.development.js` build artifact.

Reply:
that's intentional. when bundling the react app it will be tree shaken, but we want this log to show up when running in dev.
I will fix the typo